<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.39">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Working Group Leadership Team â€“ Genomic Data Science</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-e26003cea8cd680ca0c55a263523d882.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-49a30fb1c686c173be37e14c159b58c8.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">Genomic Data Science</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="./about.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./about-team.html" aria-current="page"> 
<span class="menu-text">Leadership Team</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./meeting_schedule.html"> 
<span class="menu-text">Meeting Schedule</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/jhugenomicdatascience"> <i class="bi bi-github" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#stephanie-hicks" id="toc-stephanie-hicks" class="nav-link active" data-scroll-target="#stephanie-hicks">Stephanie Hicks</a></li>
  <li><a href="#kasper-hansen" id="toc-kasper-hansen" class="nav-link" data-scroll-target="#kasper-hansen">Kasper Hansen</a></li>
  <li><a href="#hongkai-ji" id="toc-hongkai-ji" class="nav-link" data-scroll-target="#hongkai-ji">Hongkai Ji</a></li>
  <li><a href="#margaret-taub" id="toc-margaret-taub" class="nav-link" data-scroll-target="#margaret-taub">Margaret Taub</a></li>
  <li><a href="#weiqiang-zhou" id="toc-weiqiang-zhou" class="nav-link" data-scroll-target="#weiqiang-zhou">Weiqiang Zhou</a></li>
  <li><a href="#ni-zhao" id="toc-ni-zhao" class="nav-link" data-scroll-target="#ni-zhao">Ni Zhao</a></li>
  <li><a href="#ingo-ruczinski" id="toc-ingo-ruczinski" class="nav-link" data-scroll-target="#ingo-ruczinski">Ingo Ruczinski</a></li>
  <li><a href="#nilanjan-chatterjee" id="toc-nilanjan-chatterjee" class="nav-link" data-scroll-target="#nilanjan-chatterjee">Nilanjan Chatterjee</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Working Group Leadership Team</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>The organizers of the working group are the Biostats faculty working in genomics, including</p>
<section id="stephanie-hicks" class="level2">
<h2 class="anchored" data-anchor-id="stephanie-hicks">Stephanie Hicks</h2>
<div>

</div>
<div class="profile-header quarto-layout-panel" data-layout="[21,79]">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 21.0%;justify-content: center;">
<p><img src="images/stephanie_hicks.png" class="profile-pic img-fluid"></p>
</div>
<div class="quarto-layout-cell" style="flex-basis: 79.0%;justify-content: flex-start;">
<p><a href="https://www.stephaniehicks.com"><strong>Stephanie Hicks</strong></a> | Hello! I work at the intersection of data science and health as an Associate Professor of Biostatistics and Biomedical Engineering at Johns Hopkins University. I develop computational methods using statistics and machine learning for the analysis of single-cell and spatial genomic and imagine data. In 2024, I co-founded the JHU Genomic Data Science Working Group and today I help co-organize events! I work to made these spaces more welcoming, diverse, and inclusive. ðŸŒ±</p>
</div>
</div>
</div>
</section>
<section id="kasper-hansen" class="level2">
<h2 class="anchored" data-anchor-id="kasper-hansen">Kasper Hansen</h2>
<div>

</div>
<div class="profile-header quarto-layout-panel" data-layout="[21,79]">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 21.0%;justify-content: center;">
<p><img src="images/kasper_hansen.png" class="profile-pic img-fluid"></p>
</div>
<div class="quarto-layout-cell" style="flex-basis: 79.0%;justify-content: flex-start;">
<p><a href="https://www.hansenlab.org"><strong>Kasper Hansen</strong></a> | I work on Genomic Data Science. My lab develops computational methods for analysis of high-throughput genomic data including transcriptomics and epigenomics measured using bulk, single-cell and spatial technologies. I have developed a number of very widely used software packages primarily released through the Bioconductor project. I am an associate professor of Biostatistics, Biomedical Engineering and Genetic Medicine.</p>
</div>
</div>
</div>
</section>
<section id="hongkai-ji" class="level2">
<h2 class="anchored" data-anchor-id="hongkai-ji">Hongkai Ji</h2>
<div>

</div>
<div class="profile-header quarto-layout-panel" data-layout="[21,79]">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 21.0%;justify-content: center;">
<p><img src="images/hongkai_ji.jpg" class="profile-pic img-fluid"></p>
</div>
<div class="quarto-layout-cell" style="flex-basis: 79.0%;justify-content: flex-start;">
<p><a href="https://jilab.org"><strong>Hongkai Ji</strong></a> | My team specializes in developing statistical, computational, and machine learning methods, along with software tools, for high-throughput omics data. This includes bulk, single-cell, and spatial transcriptomics, epigenomics, and immune repertoire data. We focus on creating methods for data denoising, multi-omics analysis, managing data with complex structures, scalable data integration, comparative functional genomics, and genome-wide cross-modality prediction. Our methodological research is closely intertwined with the development of new technologies. Additionally, we apply these methods and tools to decode gene regulation, study cancer and cancer immunotherapy, investigate infectious diseases, and explore the early life origins of human diseases.</p>
</div>
</div>
</div>
</section>
<section id="margaret-taub" class="level2">
<h2 class="anchored" data-anchor-id="margaret-taub">Margaret Taub</h2>
<div>

</div>
<div class="profile-header quarto-layout-panel" data-layout="[21,79]">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 21.0%;justify-content: center;">
<p><img src="images/margaret_taub.png" class="profile-pic img-fluid"></p>
</div>
<div class="quarto-layout-cell" style="flex-basis: 79.0%;justify-content: flex-start;">
<p><a href="https://publichealth.jhu.edu/faculty/2744/margaret-taub"><strong>Margaret Taub</strong></a> | I work collaboratively with other biostatisticians, clinicians and epidemiologists to use genetic and genomic data to understand the causes of complex diseases/traits, including asthma, cardiovascular disease and chronic pain. I am particularly interested in study design and statistical approaches to alleviate technical artifacts, development of reproducible data generation and quality control pipelines, and integrative genomics data analysis, where multiple data types can be leveraged to shed light on the trait of interest.</p>
</div>
</div>
</div>
</section>
<section id="weiqiang-zhou" class="level2">
<h2 class="anchored" data-anchor-id="weiqiang-zhou">Weiqiang Zhou</h2>
<div>

</div>
<div class="profile-header quarto-layout-panel" data-layout="[21,79]">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 21.0%;justify-content: center;">
<p><img src="images/weiqiang_zhou.jpg" class="profile-pic img-fluid"></p>
</div>
<div class="quarto-layout-cell" style="flex-basis: 79.0%;justify-content: flex-start;">
<p><a href="https://www.weiqiangzhou.com"><strong>Weiqiang Zhou</strong></a> | I am a data scientist working on genomics and big data analysis. I develop statistical and computational tools for integrative analysis of large-scale genomic data, including bulk and single-cell data, to study gene regulation in different diseases and biological systems. My team currently focuses on developing machine learning (AI) models to analyze single-cell multiomics data.</p>
</div>
</div>
</div>
</section>
<section id="ni-zhao" class="level2">
<h2 class="anchored" data-anchor-id="ni-zhao">Ni Zhao</h2>
<div>

</div>
<div class="profile-header quarto-layout-panel" data-layout="[21,79]">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 21.0%;justify-content: center;">
<p><img src="images/ni_zhao.png" class="profile-pic img-fluid"></p>
</div>
<div class="quarto-layout-cell" style="flex-basis: 79.0%;justify-content: flex-start;">
<p><a href="https://www.biostat.jhsph.edu/~nzhao/"><strong>Ni Zhao</strong></a> | Our team specializes in developing and applying advanced statistical and machine learning methodologies to address challenges in large-scale â€˜omicsâ€™ studies. Our expertise spans metagenomics, genomics, immune profiling, and data derived from advanced spatial technologies, including spatial microbiome and spatial proteomics. Recent projects have focused on improving our understanding and correction of biases and unknown batch effects in microbiome research, facilitating both vertical and horizontal integration of microbiome data with other data types, and developing predictive models from high-throughput sequencing studies.</p>
</div>
</div>
</div>
</section>
<section id="ingo-ruczinski" class="level2">
<h2 class="anchored" data-anchor-id="ingo-ruczinski">Ingo Ruczinski</h2>
<div>

</div>
<div class="profile-header quarto-layout-panel" data-layout="[21,79]">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 21.0%;justify-content: center;">
<p><img src="images/ingo_ruczinski.jpg" class="profile-pic img-fluid"></p>
</div>
<div class="quarto-layout-cell" style="flex-basis: 79.0%;justify-content: flex-start;">
<p><a href="https://biostat.jhsph.edu/~iruczins/"><strong>Ingo&nbsp;Ruczinski</strong></a> | My professional expertise is in statistical genetics, genomics and proteomics. I have developed or co-developed several methodologies and open source software packages relevant for genomic array pre-processing, SNP association studies, population and family-based sequencing studies, studies of DNA copy number variants, and proteomic analyses, including data from PhIP-Seq and MIPSA experiments. I collaborate on several large-scale genomic association and sequencing studies at Johns Hopkins and as part of larger collaborations including the NHLBI Trans-Omics for Precision Medicine (TOPMed) Program.</p>
</div>
</div>
</div>
</section>
<section id="nilanjan-chatterjee" class="level2">
<h2 class="anchored" data-anchor-id="nilanjan-chatterjee">Nilanjan Chatterjee</h2>
<div>

</div>
<div class="profile-header quarto-layout-panel" data-layout="[21,79]">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 21.0%;justify-content: center;">
<p><img src="images/nilanjan_chatterjee.png" class="profile-pic img-fluid"></p>
</div>
<div class="quarto-layout-cell" style="flex-basis: 79.0%;justify-content: flex-start;">
<p><a href="https://publichealth.jhu.edu/faculty/1826/nilanjan-chatterjee"><strong>Nilanjan Chatterjee</strong></a> | Dr.&nbsp;Chatterjee is a Bloomberg Distinguished Professor at the Department of Biostatistics, Bloomberg School of Public Health, and Department of Oncology School of Medicine at the&nbsp;Johns&nbsp;Hopkins University. Before&nbsp;joining Johns Hopkins, he led the Biostatistics Branch of the Division of Cancer Epidemiology and Genetics of the US National Cancer Institute during 2008-2015. He is known for foundational and methodological contributions to multiple areas of modern biomedical data science, including large-scale analysis of genetic associations, gene-environment interactions, and predictive model building by synthesis of information from various data sources.&nbsp; He has received numerous prestigious national and international awards, including the celebrated Committee of the Presidents of Statistical Societies (COPSS) Presidentâ€™s Award. He is an elected Fellow of the American Statistical Association and American Epidemiological Society. He serves on the scientific advisory committee of the Radiation Effect Research Foundation, Hiroshima, Japan and the Population and Prevention Research Committee of the foundation of Cancer Research UK.</p>
</div>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/jhugenomicdatascience\.github\.io\/");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>