---
title: "Working Group Leadership Team"
format:
    html:
        toc: true
        toc-depth: 3
---

The organizers of the working group are the Biostats faculty working in genomics, including 

## Stephanie Hicks

::: {layout="[21,79]" .profile-header}

![](images/stephanie_hicks.png){.profile-pic}

[**Stephanie Hicks**](https://www.stephaniehicks.com) | Hello! I work at the intersection of data science and health as an Associate Professor of Biostatistics and Biomedical Engineering at Johns Hopkins University. I develop computational methods using statistics and machine learning for the analysis of single-cell and spatial genomic and imagine data. In 2024, I co-founded the JHU Genomic Data Science Working Group and today I help co-organize events! I work to made these spaces more welcoming, diverse, and inclusive. ðŸŒ±

:::

## Kasper Hansen

::: {layout="[21,79]" .profile-header}

![](images/kasper_hansen.png){.profile-pic}

[**Kasper Hansen**](https://www.hansenlab.org) | I work on Genomic Data Science. My lab develops computational methods for analysis of high-throughput genomic data including transcriptomics and epigenomics measured using bulk, single-cell and spatial technologies. I have developed a number of very widely used software packages primarily released through the Bioconductor project. I am an associate professor of Biostatistics, Biomedical Engineering and Genetic Medicine. 

:::

## Hongkai Ji

::: {layout="[21,79]" .profile-header}

![](images/hongkai_ji.jpg){.profile-pic}

[**Hongkai Ji**](https://jilab.org) | My team specializes in developing statistical, computational, and machine learning methods, along with software tools, for high-throughput omics data. This includes bulk, single-cell, and spatial transcriptomics, epigenomics, and immune repertoire data. We focus on creating methods for data denoising, multi-omics analysis, managing data with complex structures, scalable data integration, comparative functional genomics, and genome-wide cross-modality prediction. Our methodological research is closely intertwined with the development of new technologies. Additionally, we apply these methods and tools to decode gene regulation, study cancer and cancer immunotherapy, investigate infectious diseases, and explore the early life origins of human diseases.

:::

## Margaret Taub

::: {layout="[21,79]" .profile-header}

![](images/margaret_taub.png){.profile-pic}

[**Margaret Taub**](https://publichealth.jhu.edu/faculty/2744/margaret-taub) | Add here. 

:::


## Weiqiang Zhou

::: {layout="[21,79]" .profile-header}

![](images/weiqiang_zhou.jpg){.profile-pic}

[**Weiqiang Zhou**](https://www.weiqiangzhou.com) | I am a data scientist working on genomics and big data analysis. I develop statistical and computational tools for integrative analysis of large-scale genomic data, including bulk and single-cell data, to study gene regulation in different diseases and biological systems. My team currently focuses on developing machine learning (AI) models to analyze single-cell multiomics data.

:::



## Ni Zhao

::: {layout="[21,79]" .profile-header}

![](images/ni_zhao.png){.profile-pic}

[**Ni Zhao**](https://www.biostat.jhsph.edu/~nzhao/) | Our team specializes in developing and applying advanced statistical and machine learning methodologies to address challenges in large-scale â€˜omicsâ€™ studies. Our expertise spans metagenomics, genomics, immune profiling, and data derived from advanced spatial technologies, including spatial microbiome and spatial proteomics. Recent projects have focused on improving our understanding and correction of biases and unknown batch effects in microbiome research, facilitating both vertical and horizontal integration of microbiome data with other data types, and developing predictive models from high-throughput sequencing studies. 

:::


## Ingo Ruczinski

::: {layout="[21,79]" .profile-header}

![](images/ingo_ruczinski.jpg){.profile-pic}

[**IngoÂ Ruczinski**](https://biostat.jhsph.edu/~iruczins/) | My professional expertise is in statistical genetics, genomics and proteomics. I have developed or co-developed several methodologies and open source software packages relevant for genomic array pre-processing, SNP association studies, population and family-based sequencing studies, studies of DNA copy number variants, and proteomic analyses, including data from PhIP-Seq and MIPSA experiments. I collaborate on several large-scale genomic association and sequencing studies at Johns Hopkins and as part of larger collaborations including the NHLBI Trans-Omics for Precision Medicine (TOPMed) Program.

:::

## Nilanjan Chatterjee

::: {layout="[21,79]" .profile-header}

![](images/nilanjan_chatterjee.png){.profile-pic}

[**Nilanjan Chatterjee**](https://publichealth.jhu.edu/faculty/1826/nilanjan-chatterjee) | Dr. Chatterjee is a Bloomberg Distinguished Professor at the Department of Biostatistics, Bloomberg School of Public Health, and Department of Oncology School of Medicine at theÂ JohnsÂ Hopkins University. BeforeÂ joining Johns Hopkins, he led the Biostatistics Branch of the Division of Cancer Epidemiology and Genetics of the US National Cancer Institute during 2008-2015. He is known for foundational and methodological contributions to multiple areas of modern biomedical data science, including large-scale analysis of genetic associations, gene-environment interactions, and predictive model building by synthesis of information from various data sources.Â  He has received numerous prestigious national and international awards, including the celebrated Committee of the Presidents of Statistical Societies (COPSS) Presidentâ€™s Award. He is an elected Fellow of the American Statistical Association and American Epidemiological Society. He serves on the scientific advisory committee of the Radiation Effect Research Foundation, Hiroshima, Japan and the Population and Prevention Research Committee of the foundation of Cancer Research UK.

:::


